lms/djangoapps/experiments/tests/test_utils.py

```diff
@@ -116,3 +121,19 @@ def test_price_and_sku_from_multiple_courses(self):
         self.assertEqual(2, len(skus))
         self.assertIn(self.run_a_sku, skus)
         self.assertIn(self.entitlement_a_sku, skus)
+
+    def test_get_experiment_user_metadata_context(self):
+        user_metadata_expected_result = """{'username': 'robot0', 'user_id': 2, 'course_id': 'org.0/course_0/Run_0',
+            'enrollment_mode': None, 'upgrade_link': None, 'upgrade_price': 'Free', 'audit_access_deadline': None,
+            'course_duration': None, 'pacing_type': 'instructor_paced', 'has_staff_access': False, 'forum_roles': [],
+            'partition_groups': {}, 'has_non_audit_enrollments': False, 'program_key_fields': None,
+            'email': 'robot+test+0@edx.org', 'schedule_start': None, 'enrollment_time': None,
```

Comment:
i'm not sure whether some of these fields are deterministic 
you probably want to put stuff like user.email and course.start rather than the hardcoded strings

Reply:
Good call- I've updated these to use fields from the course & user objects. Two fields weren't available this way (or via 'CourseDescriptorWithMixins'): has_staff_access and has_non_audit_enrollments.